internal/archiver/archiver.go

```diff
@@ -445,36 +453,30 @@ func (arch *Archiver) Save(ctx context.Context, snPath, target string, previous
 	return fn, false, nil
 }
 
-// fileChanged returns true if the file's content has changed since the node
-// was created.
-func fileChanged(fi os.FileInfo, node *restic.Node, ignoreInode bool) bool {
-	if node == nil {
+// fileChanged tries to detect whether a file's content has changed compared
+// to the contents of node, which describes the same path in the parent backup.
+// It should only be run for regular files.
+func fileChanged(fi os.FileInfo, node *restic.Node, ignore ChangeIgnore) bool {
+	// Easy cases: type has to be "file", size and mtime must always match.
+	switch {
+	case node == nil:
 		return true
-	}
-
-	// check type change
-	if node.Type != "file" {
+	case node.Type != "file":
 		return true
-	}
-
-	// check modification timestamp
-	if !fi.ModTime().Equal(node.ModTime) {
+	case uint64(fi.Size()) != node.Size:
 		return true
```

Comment:
Is there a reason you no longer check for `uint64(extFI.Size) != node.Size`
On Linux `extFI.Size` should equal `fi.Size()`, but I'm not sure about Windows, see `internal/fs/stat_windows.go`

Reply:
On Windows it's always equal too: https://github.com/golang/go/blob/3a4322627e781c05418a8cc40ebb9f547a6c4ac8/src/os/types_windows.go#L108